server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -694,15 +693,17 @@ public void run() {
     try {
 
       final AtomicReference<Throwable> err = new AtomicReference<>();
-      final AtomicLong timeSinceLastCompletion = new AtomicLong(0L);
+      final AtomicReference<NanoTime> lastCompletionTime = new AtomicReference<>(null);
 
       while (!shutdown) {
 
         idleProcessCheck(() -> {
-          return timeSinceLastCompletion.get() == 0
-              /* Never started a compaction */ || (timeSinceLastCompletion.get() > 0
```

Comment:
These changes fix a bug in the existing code. Seems like the existing code was assuming that nano times would be positive which I do not think is safe to assume.   This change is being made in main, if the bug exists in 2.1 then may want to fix it in another PR w/o the NanoTime changes.